### PR TITLE
Add an env variable to taint windows nodes

### PR DESCRIPTION
**Pull request by yujuhong**

### CODE REVIEW

**Comment by pjh** [comment]
LGTM

**Comment by yujuhong** [comment]
Now that we're tainting windows nodes by default, we have to untaint them in our test script. Update the PR to do so

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -843,6 +843,11 @@ function construct-windows-kubelet-flags {
     flags+=" --node-labels=${node_labels}"
   fi
 
+  # Apply Windows-only node taints.
+  if [[ -n "${WINDOWS_NODE_TAINTS:-}" ]]; then
+    flags+=" --register-with-taints=${WINDOWS_NODE_TAINTS}"
```
Comment (pjh) [inline cluster/gce/util.sh]: At line 758 we're already setting --register-with-taints with the common (Linux and Windows) NODE_TAINTS. Can you move this flag out of construct-common-kubelet-flags and set it correctly here and in construct-linux-kubelet-flags?

Even if setting the flag twice does what we want (applies both sets of node taints), it's confusing.

File: hack/run-e2e.sh [hunk 2]
```diff
@@ -19,14 +19,17 @@ for node in $LINUX_NODES; do
   kubectl taint node $node node-under-test=false:NoSchedule
 done
 
+# Untaint the windows nodes to allow test workloads without tolerations to be
+# scheduled onto them.
+WINDOWS_NODES=$(kubectl get nodes -l beta.kubernetes.io/os=windows -o name)
+for node in $WINDOWS_NODES; do
+  kubectl taint node $node node.kubernetes.io/os:NoSchedule-
+done
```
Comment (pjh) [inline hack/run-e2e.sh]: echo a log message here?

**Comment by pjh** [comment]
Thanks, LGTM.

**Comment by yujuhong** [comment]
Verified that the cluster was up and the nodes were tainted correctly.
PTAL again

**Comment by pjh** [comment]
Doing this logic in bash is so painful <eyeroll>

But LGTM!

**Pull request merged**